src/lightning/fabric/strategies/model_parallel.py

```diff
@@ -0,0 +1,331 @@
+        storage_options: Optional[Any] = None,
+        filter: Optional[Dict[str, Callable[[str, Any], bool]]] = None,
+    ) -> None:
+        """Save model, optimizer, and other state to a checkpoint on disk."""
+        if storage_options is not None:
+            raise TypeError(
+                f"`{self.__class__.__name__}.save_checkpoint(..., storage_options=...)` is not supported because"
+                f" `{self.__class__.__name__}` does not use the `CheckpointIO`."
+            )
+        if filter is not None:
+            raise NotImplementedError(f"{self.__class__.__name__} does not yet support the `filter` argument.")
+
+        # broadcast the path from rank 0 to ensure all the states are saved in a common path
+        path = Path(self.broadcast(path))
+        _distributed_checkpoint_save(state, path)
+
+    @override
+    def load_checkpoint(
+        self,
+        path: _PATH,
+        state: Optional[Union[Module, Optimizer, Dict[str, Union[Module, Optimizer, Any]]]] = None,
+        strict: bool = True,
+    ) -> Dict[str, Any]:
+        if isinstance(state, (Module, Optimizer)):
+            raise NotImplementedError(
+                "Loading a module or optimizer object from a checkpoint directly is not yet supported."
+            )
+        if strict is False:
+            raise NotImplementedError(f"Non-strict loading is not yet supported in {self.__class__.__name__}.")
+
+        # broadcast the path from rank 0 to ensure all the states are loaded from a common path
+        path = Path(self.broadcast(path))
+        _distributed_checkpoint_load(state, path)  # type: ignore[arg-type]
+        return {}
+
+    def _setup_distributed(self) -> None:
+        reset_seed()
+        self._set_world_ranks()
+        self._process_group_backend = self._get_process_group_backend()
+        assert self.cluster_environment is not None
+        _init_dist_connection(self.cluster_environment, self._process_group_backend, timeout=self._timeout)
+
+    def _setup_device_mesh(self) -> None:
+        from torch.distributed.device_mesh import init_device_mesh
+
+        if self._data_parallel_size == "auto":
+            self._data_parallel_size = self.num_nodes
+        if self._tensor_parallel_size == "auto":
+            self._tensor_parallel_size = self.num_processes
+        if self._data_parallel_size * self._tensor_parallel_size != self.world_size:
+            raise RuntimeError(
+                f"The sizes `data_parallel_size={self._data_parallel_size}` and"
+                f" `tensor_parallel_size={self._tensor_parallel_size}` multiplied should equal the world size"
+                f" ({self.world_size})."
+            )
+        self._device_mesh = init_device_mesh(
+            device_type=self.root_device.type,
+            mesh_shape=(self._data_parallel_size, self._tensor_parallel_size),
+            mesh_dim_names=("data_parallel", "tensor_parallel"),
+        )
```

Comment:
What about having this be the strategy API?:

```
parallelize_fn=...
device_mesh={"data_parallel": "auto", "tensor_parallel": "auto"}
data_parallel_dim_name="data_parallel", # default
tensor_parallel_dim_name="tensor_parallel",  # default
```

Reply:
I don't like that this is a dict with arbitrary names. My intention was to enforce a clear contract. This is like proposing:

```
Trainer(args={"devices": 2}, default_devices_name="devices") 
```
I don't see why the user would need this configuration.